actionpack/lib/action_controller/metal/request_forgery_protection.rb

```diff
@@ -213,7 +213,13 @@ def verify_authenticity_token
 
         if !verified_request?
           if logger && log_warning_on_csrf_failure
-            logger.warn "Can't verify CSRF token authenticity."
+            if valid_request_origin?
+              logger.warn "Can't verify CSRF token authenticity."
+            else
+              logger.warn "Can't verify CSRF token authenticity as the " \
+                "request origin header(#{request.origin}) doesn't match" \
```

Comment:
You want a space before the `(` on this line, and the same on the one below. Also, you'll want a space at the end of this line.